packages/palette/src/elements/FilterSelect/Components/FilterSelectContext.tsx

```diff
@@ -148,8 +160,23 @@ export const FilterSelectContextProvider: React.FC<
         payload: { query },
       })
     },
+    setSelectedItems: (items) => {
+      dispatch({
+        type: "SET_SELECTED_ITEMS",
+        payload: {
+          items,
+        },
+      })
+    }
   }
 
+  useEffect(() => {
+    if (props.selectedItems?.length) {
+      contextValue.setSelectedItems(props.selectedItems)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [props.selectedItems?.length])
```

Comment:
**question**: @dzucconi can we install and use `use-deep-compare-effect` for this case?

Reply:
https://bundlephobia.com/package/use-deep-compare-effect@1.8.1 is probably preferable over lodash. But lodash is already in here for this component alone so I suppose either is fine.

Reply:
Either or is fine! checking the length of the array is fine though for this.

Reply:
Got it. Thanks for the feedback